### PR TITLE
adds topological sort example

### PR DESCRIPTION
**Pull request by SeeThruHead**
add topological sort to new examples folder
adds examples readme entry with link to specific example
gitignore example output files

prior work: #79

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
   "dependencies": {
     "fs-extra": "^0.30.0",
     "lodash": ">=3.5 <5",
-    "mutexify": "1.0.1"
+    "mutexify": "1.0.1",
+    "toposort": "^1.0.3"
```
Comment (mastilver) [inline package.json]: Can you move that to `devDependencies`

File: examples/topological-sort/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+module.exports = {
+  context: __dirname,
+  entry: {
+    main: './main.js',
+    vendor: './util.js'
+  },
+  output: {
+    filename: '[name].out.js'
```
Comment (mastilver) [inline examples/topological-sort/webpack.config.js]: can you set `output.path` to `path.join(__dirname, 'build')` and `filename` to `[name].js`

I've got plan for the future to use those in tests, that will make it easier

File: examples/topological-sort/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+var plugin = require('../../index.js');
+var webpack = require('webpack');
+var _ = require('lodash');
+var toposort = require('toposort');
+
+function generate(seed, files) {
+  var nodeMap = {};
+
+  files.forEach(function (file) {
+    nodeMap[file.chunk.id] = file;
```
Comment (joscha) [inline examples/topological-sort/webpack.config.js]: I think this needs an `isChunk` guard.

File: examples/topological-sort/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  });
+
+  var edges = [];
+
+  files.forEach(function (file) {
+    if (file.chunk.parents) {
```
Comment (joscha) [inline examples/topological-sort/webpack.config.js]: same here

**Comment by SeeThruHead** [comment]
i can take a look at this tmrw. what better way to spend my airport wait :D

File: examples/topological-sort/webpack.config.js (hunk 4 above)
Comment (SeeThruHead) [reply examples/topological-sort/webpack.config.js]: so i'm not actually all that familiar with webpack, what kind of dependency tree would give me files without chunks? I supposed the example should handle that case? or is it sufficient to handle the actual case that the example files dep tree creates (all files are chunks?)

**Comment by mastilver** [comment]
Hey @SeeThruHead :)

Let me know how you are getting on with that. Do you want me to take it over (I will have more time in a couple of weeks)?

**Comment by SeeThruHead** [comment]
hey @mastilver I made a comment above asking for some guidance with how to test an isChunk guard. ie what kind of dep tree would give me files without chunks.

File: examples/topological-sort/webpack.config.js (hunk 4 above)
Comment (mastilver) [reply examples/topological-sort/webpack.config.js]: Oops sorry, missed that comment.

We include emitted files (for example from `copy-webpack-plugin`), basically it's every non-js files. I think those better be at the end (but really I don't care, up to you!)

**Comment by caub** [comment]
it's a bit brutal, but I've done it here https://github.com/caub/webpack-manifest-plugin/blob/m2/lib/plugin.js (the logic actually mostly copied from HtmlWebpackPlugin)

I think the manifest file deserves to be sorted by default

**Comment by mastilver** [comment]
html-webpack-plugin is getting rid of toposort, in flavours of `entrypoints`, we should do the same for this plugin I believe

**Comment by shellscape** [comment]
@SeeThruHead Thanks for opening this PR a while back, and I'm sorry that it didn't get attention sooner. We've just landed a major refactor in #222, and part of that is only supporting webpack v4.44.0+. Please let me know if you're interested in reopening this PR and we'll move forward.

**Pull request closed without merging**